hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -427,6 +428,52 @@ public static boolean isMetadataOnlyUpdateOp(@NonNull final TokenUpdateTransacti
         return op.equals(defaultWithMetadata);
     }
 
+    /**
+     * Check if TokenUpdateOp wants to update only some of the low priority keys or the metadata field
+     * low priority keys are -> wipeKey, kycKey, supplyKey, freezeKey, feeScheduleKey, pauseKey or metadataKey
+     */
+    public static boolean isAdminOrLowPriorityKeyRemoval(
+            @NonNull final TokenUpdateTransactionBody op, @NonNull final Token originalToken) {
+        return originalToken.hasAdminKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.adminKey())
+                || originalToken.hasWipeKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.wipeKey())
+                || originalToken.hasKycKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.kycKey())
+                || originalToken.hasSupplyKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.supplyKey())
+                || originalToken.hasFreezeKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.freezeKey())
+                || originalToken.hasFeeScheduleKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.feeScheduleKey())
+                || originalToken.hasMetadataKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.pauseKey())
+                || originalToken.hasMetadataKey() && IMMUTABILITY_SENTINEL_KEY.equals(op.metadataKey());
```

Comment:
Checks whether `originalToken` has some key are not necessary because [here](https://github.com/hashgraph/hedera-services/blob/12158-enable-admin-key-to-remove-all-keys/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenUpdateValidator.java#L65) we make them and will indicate with `TOKEN_IS_IMMUTABLE`. We can replace them with checks if `op` has some key to ensure it is not a `null`.
Also `IMMUTABILITY_SENTINEL_KEY.equals(someKey)` can be replaced with `isKeyRemoval(someKey)`

_All of these apply to all key occurrences_